### PR TITLE
Domain notify resync period

### PR DESCRIPTION
**Pull request by davidvossel**
If we miss a domain update from virt-launcher for any reason, that causes issues such as never reporting the status of a migration. This patch introduces a periodic resync that forces virt-handler to update it's virt-launcher cache every 5 minutes by default. The period is configurable. The goal here is to be resilient to missed virt-launcher notifications. 

I also restructured how the tickers work in the domain informer to ensure we always call "Stop()" on tickers. This theoretically could close a resource leak. It's unclear how the garbage collector handles tickers that are never stopped, yet never referenced.  

```release-note
Adds recurring period that resyncs virt-launcher domains with virt-handler
```

### CODE REVIEW

File: pkg/virt-handler/cache/cache.go [hunk 1]
```diff
@@ -299,10 +301,18 @@ func (d *DomainWatcher) startBackground() error {
...
+		resyncTicker := time.NewTicker(d.resyncPeriod)
+		resyncTickerChan := resyncTicker.C
+		defer resyncTicker.Stop()
+
 		// Divide the watchdogTimeout by 3 for our ticker.
 		// This ensures we always have at least 2 response failures
 		// in a row before we mark the socket as unavailable (which results in shutdown of VMI)
-		expiredWatchdogTicker := time.NewTicker(time.Duration((d.watchdogTimeout/3)+1) * time.Second).C
+		expiredWatchdogTicker := time.NewTicker(time.Duration((d.watchdogTimeout/3)+1) * time.Second)
+		defer expiredWatchdogTicker.Stop()
```
Comment (danielBelenky) [inline pkg/virt-handler/cache/cache.go:312]: Is this change related to the PR? or you just closed a leak that we had? It's it's the latter, could you mention it in the PR description?

Comment (davidvossel) [reply pkg/virt-handler/cache/cache.go:312]: done

File: pkg/virt-handler/cache/cache.go [hunk 2]
```diff
@@ -348,6 +360,41 @@ func (d *DomainWatcher) handleStaleWatchdogFiles() error {
...
+		defer client.Close()
+
+		domain, exists, err := client.GetDomain()
+		if err != nil {
+			// this resync is best effort only.
+			log.Log.Reason(err).Errorf("unable to retrieve domain at socket %s during resync", socket)
+			continue
+		} else if !exists {
+			// nothing to sync if it doesn't exist
+			continue
```
Comment (rmohr) [inline pkg/virt-handler/cache/cache.go:391]: We still get the final delete notification somewhere else in this case when the socketFile is gone?

Comment (rmohr) [reply pkg/virt-handler/cache/cache.go:391]: These two do that, right?

```
				d.handleStaleWatchdogFiles()
				d.handleStaleSocketConnections()
```

**Comment by rmohr** [comment]
/lgtm

The existing test actually starts the server and verifies that it will happen. Good enough for me.

File: pkg/virt-handler/cache/cache.go (hunk 2 above)
Comment (davidvossel) [reply pkg/virt-handler/cache/cache.go:391]: yup. so it's safe to ignore connection errors here. This resync is best effort, and the stale socket connections function handles the rest.

**Comment by vladikr** [comment]
Make sense to me :+1: 
/lgtm
